storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManagerConfig.java

```diff
@@ -257,13 +257,13 @@ public static ConfigDef configDef() {
                         atLeast(1),
                         MEDIUM,
                         REMOTE_LOG_MANAGER_THREAD_POOL_SIZE_DOC)
-                .defineInternal(REMOTE_LOG_MANAGER_COPIER_THREAD_POOL_SIZE_PROP,
+                .define(REMOTE_LOG_MANAGER_COPIER_THREAD_POOL_SIZE_PROP,
                         INT,
                         DEFAULT_REMOTE_LOG_MANAGER_COPIER_THREAD_POOL_SIZE,
```

Comment:
Unrelated to this change, but the KIP mentions:

> If no thread values are set for the two new configurations presented later on in the document we will default to using the same number of threads in each pool as detailed by remote.log.manager.thread.pool.size.

Perhaps I'm mistaken, but this gives the impression that the default for the new config is falling back to `remote.log.manager.thread.pool.size`. [Configuration section](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=263424654#KIP950:TieredStorageDisablement-Configuration) however specifies a default of 10 for the new configs. Seems a bit ambiguous and I wonder if the default is correct?

Reply:
IMO it doesn't make sense to fallback to a deprecated property, so I think we should leave the current settings, and maybe update the KIP. @showuon wdyt?

Reply:
@gaurav-narula I see the backwards compatibility issue now. Let's do as @showuon suggests. I'm also updating the doc to make it explicit. Please check the latest commit.

Reply:
+1 not to break backward compatibility here. 

But it is better to go with default value as -1 and the validation can be -1 or >0.

We can clarify the doc adding with the below text for better clarity.
"The default value of -1 means that this will be set to the configured value of `remote.log.manager.thread.pool.size`, if available; otherwise, it defaults to 10" .